tests/integration/test_hyper_transformer.py

```diff
@@ -3,7 +3,80 @@
 import pytest
 
 from rdt import HyperTransformer
-from rdt.transformers import OneHotEncodingTransformer
+from rdt.transformers import (
+    BaseTransformer, BooleanTransformer, CategoricalTransformer, NumericalTransformer,
+    OneHotEncodingTransformer)
+
+
+class DummyTransformerNumerical(BaseTransformer):
+
+    INPUT_TYPE = 'categorical'
+    OUTPUT_TYPES = {
+        'value': 'float'
+    }
+
+    def _fit(self, data):
+        pass
+
+    def _transform(self, data):
+        return data.astype(float)
+
+    def _reverse_transform(self, data):
+        return data.astype(str)
+
+
+class DummyTransformerNotMLReady(BaseTransformer):
+
+    INPUT_TYPE = 'datetime'
+    OUTPUT_TYPES = {
+        'value': 'categorical',
+    }
+
+    def _fit(self, data):
+        pass
+
+    def _transform(self, data):
+        # Stringify input data
+        return data.astype(str)
+
+    def _reverse_transform(self, data):
+        return data.astype('datetime64')
+
+
+class DummyTransformerMultiColumn(BaseTransformer):
+
+    INPUT_TYPE = 'datetime'
+    OUTPUT_TYPES = {
+        'value': 'float',
+    }
+
+    def _fit(self, data):
+        pass
+
+    def _transform(self, data):
+        # Convert multiple columns into a single datetime
+        data.columns = [c.replace('_str.value', '') for c in data.columns]
+        data = pd.to_datetime(data)
+
+        out = pd.DataFrame(dict(zip(
+            self.output_columns,
+            [
+                data.values.astype(np.float64),
+                data.isnull().astype(np.float64)
+            ]
+        ))).fillna(-1)
+
+        return out
+
+    def _reverse_transform(self, data):
+        datetimes = data.round().astype('datetime64[ns]')
+        out = pd.DataFrame({
+            'year': datetimes.dt.year,
+            'month': datetimes.dt.month,
+            'day': datetimes.dt.day,
+        })
+
+        return out
 
 
 def get_input_data_with_nan():
```

Comment:
I'm thinking that we should complicate things a bit more by making this DataFrame use an index that is not the default 0, 1, 2, 3.
I suggest adding an `index` argument and pass something random, so we validate that the operations below do not mess the index.

Reply:
After discussion, we will open another issue for this because there does seem to be an indexing error in the `BaseTransformer` class